dynamiq/memory/backends/qdrant.py

```diff
@@ -13,46 +15,24 @@
 
 class QdrantError(Exception):
     """Base exception for Qdrant-related errors."""
-
     pass
 
 
 class Qdrant(MemoryBackend):
     """Qdrant implementation of the memory storage backend."""
 
-    name = "Qdrant"
-
-    def __init__(
-        self,
-        connection: QdrantConnection,
-        embedder: BaseEmbedder,
-        index_name: str = "conversations",
-        metric: str = "cosine",
-        on_disk: bool = False,
-        create_if_not_exist: bool = True,
-    ):
-        """Initializes the Qdrant memory storage.
-
-        Args:
-            connection: QdrantConnection instance for connecting to Qdrant
-            embedder: Embedder instance for creating embeddings
-            index_name: Name of the collection to store messages
-        """
-        self.connection = connection
-        self.index_name = index_name
-        self.embedder = embedder
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+
+    name: str = "Qdrant"
+    connection: QdrantConnection
+    embedder: BaseEmbedder
+    vector_store: QdrantVectorStore
+    _client: QdrantClient | None = None
 
+    def model_post_init(self, __context) -> None:
+        """Initialize the vector store after model initialization."""
         try:
-            self.vector_store = QdrantVectorStore(
-                connection=connection,
-                index_name=index_name,
-                dimension=embedder.dimensions,
-                create_if_not_exist=create_if_not_exist,
-                metric=metric,
-                on_disk=on_disk,
-                recreate_index=False,
-            )
-            self.client = self.vector_store._client
+            self._client = self.vector_store._client
             if not self.client:
```

Comment:
Defined `self._client` in prev line, so make fix to use it

Reply:
Let's make single raise with proper message, instead of raise, catch and extra raise